src/ui/components/Library/RecordingOptionsDropdown.tsx

```diff
@@ -8,28 +8,47 @@ import MaterialIcon from "../shared/MaterialIcon";
 import hooks from "ui/hooks";
 import { RecordingId } from "@recordreplay/protocol";
 import { WorkspaceId } from "ui/state/app";
-import { Dropdown, DropdownDivider, DropdownItem } from "./LibraryDropdown";
+import { Dropdown, DropdownItem } from "./LibraryDropdown";
 import PortalDropdown from "../shared/PortalDropdown";
 import classNames from "classnames";
 import MoveRecordingMenu from "./MoveRecordingMenu";
 import { useConfirm } from "../shared/Confirm";
 import { isPublicDisabled } from "ui/utils/org";
+import { useGetNonPendingWorkspaces } from "ui/hooks/workspaces";
 
 type RecordingOptionsDropdownProps = PropsFromRedux & {
   recording: Recording;
 };
 
+function useGetShowDropdown(recording: Recording) {
+  const { userId, loading } = hooks.useGetUserId();
+  const { workspaces } = useGetNonPendingWorkspaces();
+
+  if (loading) {
+    return { showDropdown: false, loading };
+  }
+
+  const isOwner = userId == recording.user?.id;
+  // Node recordings don't have an owner since they're uploaded using the workspace's
+  // API key. We add this check here so the team is able to access the dropdown.
+  const sameTeam =
+    recording.workspace?.id && workspaces.find(w => w.id === recording.workspace?.id);
+  const showDropdown = isOwner || (!recording.user && sameTeam);
+
+  return { showDropdown, loading };
+}
```

Comment:
This is actually checking belonging/ownership. The name of the hook can be something like `useOwnershipInfo` or so.

The fact that this info is being used to show/hide a dropdown is less relevant to the actual job of the hook and it would be good to decouple those two.